tests/test_black.py

```diff
@@ -1303,6 +1303,28 @@ def test_read_pyproject_toml(self) -> None:
         self.assertEqual(config["exclude"], r"\.pyi?$")
         self.assertEqual(config["include"], r"\.py?$")
 
+    def test_black_replace_config(self) -> None:
+        test_toml_file = THIS_DIR / "test_replace.toml"
+        fake_ctx = FakeContext()
+        black.read_pyproject_toml(fake_ctx, FakeParameter(), str(test_toml_file))
+        config = fake_ctx.default_map
+        # `0` in `test_replace.toml` and `1` in `_black_config.toml`.
+        # Should be `0` as the root config is on a higher level than linked configs
+        self.assertEqual(config["verbose"], "0")
+        self.assertEqual(config["check"], "no")
+        self.assertEqual(config["diff"], "y")
+        self.assertEqual(config["color"], "False")
+        self.assertEqual(config["line_length"], "88")
+        self.assertEqual(config["target_version"], ["py36", "py37", "py38"])
+        self.assertEqual(config["exclude"], r"\.pyi?$")
+        self.assertEqual(config["include"], r"\.py?$")
+
+    def test_invalid_black_config(self) -> None:
+        test_toml_file = THIS_DIR / "invalid_test.toml"
+        fake_ctx = FakeContext()
+        with self.assertRaises(click.exceptions.FileError):
+            black.read_pyproject_toml(fake_ctx, FakeParameter(), str(test_toml_file))
+
```

Comment:
Could you add a unit test where two toml files mutually include each other? Unless I'm misreading that would currently cause infinite recursion; in any case it should be tested.